File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        Set<Integer> x=new HashSet();
        for(int a:nums){
if(x.contains(a)){
    return a;
}else{
    x.add(a);
}
        
    }return -1;
}

}

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.js
var findDuplicate = function(nums) {
    //just practising in javascript(violating the condition by sorting)
nums.sort();

for (let i = 0; i < nums.length; i+=2) {
    if (nums[i] === nums[i+1] || nums[i] === nums[i-1]) 
        return nums[i];
}
};